main/networking/mqtt/core_mqtt_agent_manager.c

```diff
@@ -586,6 +586,9 @@ static MQTTStatus_t prvCoreMqttAgentInit( NetworkContext_t * pxNetworkContext )
     xTransport.send = espTlsTransportSend;
     xTransport.recv = espTlsTransportRecv;
 
+    vTlsSetConnectTimeout( 3000 );
+    vTlsSetRecvTimeout( 100 );
```

Comment:
If we are setting the timeout to 100, would this not stall the MQTT Agent?
As far as I understand, MQTT Agent is supposed to be used with non blocking sockets for receiving data.